experimenter/experimenter.py

```diff
@@ -64,14 +65,21 @@ def __init__(self, datasets_dir: str, volumes_dir: str, input_problem_directory=
         if generate_pipelines:
             print("Generating pipelines...")
             self.generated_pipelines: dict = self.generate_pipelines(self.preprocessors, self.models)
+            # TODO: create a system for how to use the next line
+            # self.generated_pipelines: dict = self.generate_k_ensembles(k_ensembles=3, p_preprocessors=2,
+            #                                                            n_generated_pipelines=1, same_model=False,
+            #                                                            same_preprocessor_order=False)
             self.num_pipelines = len(self.generated_pipelines["classification"]) + \
                                  len(self.generated_pipelines["regression"])
 
             print("There are {} pipelines".format(self.num_pipelines))
 
             if location is None:
+                self.mongo_database = PipelineDB()
                 print('Exporting pipelines to mongodb...')
                 self.output_pipelines_to_mongodb()
+            elif location == "test":
+                pass
```

Comment:
This does not seem like good practice to have test related code in real code. You can set location to a specific place and then clean it up after the test is over.

Reply:
The problem is that one of the test cases in test_generate_pipelines was creating a pipeline and then submitting it to mongodb.  I wanted to have an option to generate pipelines but not store them anywhere.  Perhaps a better name would be "NoStore" or something like that.  We could pass it a location for the test, but I thought that case deserved it's own option.  I'm fine either way though.

Reply:
> The problem is that one of the test cases in test_generate_pipelines was creating a pipeline and then submitting it to mongodb.

Reply:
Sure that works!  I just thought maybe we wanted an option to generate but not store right away.  I can't think of many uses offhand, so we can definitely go that route

Reply:
Ideally (I think) we would not be writing to any location in the constructor. Rather, we would have methods to write to database or to write to file.

Reply:
Yes, I agree.  Do you want me to refactor it for this PR?

Reply:
Let's not do the major refactoring for this PR. Let's just write the files to /dev/null for tests

Reply:
Do I need to change anything in this PR for this?  I'm trying to get this ready so we can merge it in

Reply:
Yes, let's remove `location == "test"`. Us the path "/dev/null" in the test code when you don't want it to go anywhere.